pootle/i18n/dates.py

```diff
@@ -35,14 +37,18 @@ def locale_code(self):
         return system_locale.getlocale()[0]
 
     def format_timesince(self, timestamp, locale=None):
-        return format_timedelta(
-            datetime.now()
-            - datetime.fromtimestamp(
-                timestamp),
-            locale=(
-                locale
-                or self.locale_code
-                or self.default_locale))
+        timedelta = datetime.now() - datetime.fromtimestamp(timestamp)
+        locale = (locale
+                  or self.locale_code
+                  or self.default_locale)
+        try:
+            return format_timedelta(timedelta, locale=locale)
+        except UnknownLocaleError:
+            locale = settings.LANGUAGE_CODE.split("-")[0]
```

Comment:
Definitely a no no.  I'm pretty sure I used something that can give you a valid code.